app/src/main/java/org/jellyfin/androidtv/util/apiclient/ApiClientExtensions.kt

```diff
@@ -29,3 +30,25 @@ suspend fun ApiClient.getItem(id: String): BaseItemDto? = suspendCoroutine { con
 		override fun onError(exception: Exception?) = continuation.resume(null)
 	})
 }
+
+suspend fun ApiClient.markPlayed(itemId: String, userId: String, datePlayed: Date?) : UserItemDataDto? = suspendCoroutine { continuation ->
+	MarkPlayedAsync(itemId, userId, datePlayed, object : Response<UserItemDataDto>() {
+		override fun onResponse(response: UserItemDataDto?) { continuation.resume(response!!) }
+		override fun onError(exception: Exception?) { continuation.resume(null) }
+	})
+}
+
+suspend fun ApiClient.markUnplayed(itemId: String, userId: String) : UserItemDataDto? = suspendCoroutine { continuation ->
+	MarkUnplayedAsync(itemId, userId, object : Response<UserItemDataDto>() {
+		override fun onResponse(response: UserItemDataDto?) { continuation.resume(response!!) }
+		override fun onError(exception: Exception?) { continuation.resume(null) }
+	})
+}
+
+
+suspend fun ApiClient.updateFavoriteStatus(itemId: String, userId: String, isFavorite: Boolean) : UserItemDataDto? = suspendCoroutine { continuation ->
+	UpdateFavoriteStatusAsync(itemId, userId, isFavorite, object : Response<UserItemDataDto>() {
+		override fun onResponse(response: UserItemDataDto?) { continuation.resume(response!!) }
+		override fun onError(exception: Exception?) { continuation.resume(null) }
```

Comment:
Maybe we should add proper error handling for all these coroutines before we release

Reply:
I'll make a note in the issue for that.